packages/core/src/state-management/repository.ts

```diff
@@ -180,17 +181,35 @@ export class Repository {
     opts = { ...DEFAULT_LOAD_OPTS, ...opts }
 
     const [state$, synced] = await this.loadingQ.forStream(streamId).run(async () => {
-      const [streamState$, alreadySynced] = await this._loadGenesis(streamId)
-      if (opts.sync == SyncOptions.PREFER_CACHE && alreadySynced) {
-        return [streamState$, alreadySynced]
-      }
-
-      if (opts.sync == SyncOptions.NEVER_SYNC) {
-        return [await this.stateManager.verifyLoneGenesis(streamState$), alreadySynced]
+      switch (opts.sync) {
+        case SyncOptions.PREFER_CACHE: {
+          const [streamState$, alreadySynced] = await this._loadGenesis(streamId)
+          if (alreadySynced) {
+            return [streamState$, alreadySynced]
+          } else {
+            await this.stateManager.sync(streamState$, opts.syncTimeoutSeconds * 1000)
+            return [await this.stateManager.verifyLoneGenesis(streamState$), true]
+          }
+        }
+        case SyncOptions.NEVER_SYNC: {
+          const [streamState$, alreadySynced] = await this._loadGenesis(streamId)
+          return [await this.stateManager.verifyLoneGenesis(streamState$), alreadySynced]
+        }
+        case SyncOptions.SYNC_ALWAYS: {
+          const [fromNetwork$, fromMemoryOrStore] = await Promise.all([
```

Comment:
```suggestion
          // When SYNC_ALWAYS is provided, we want to reapply and re-validate
          // the stream state.  We effectively throw out our locally stored state
          // as its possible that the commits that were used to construct that
          // state are no longer valid (for example if the CACAOs used to author them
          // have expired since they were first applied to the cached state object).
          // But if we were the only node on the network that knew about the most
          // recent tip, we don't want to totally forget about that, so we pass the tip in 
          // to `sync` so that it gets considered alongside whatever tip we learn
          // about from the network.
          const [fromNetwork$, fromMemoryOrStore] = await Promise.all([
```